packages/koa-metrics/src/index.ts

```
@@ -0,0 +1,66 @@
+/* eslint-disable camelcase */
+import {StatsD} from 'hot-shots';
+import {Context} from 'koa';
+
+export interface Options {
+  prefix: string;
+  host: string;
+}
+
+export default function metrics({prefix, host}: Options) {
+  const statsdHost = host.split(':')[0];
+  const statsdPort = parseInt(host.split(':')[1], 10);
+
+  return async function statsdMiddleware(ctx: Context, next: Function) {
+    const requestStartTime = process.hrtime();
+    // @ts-ignore the types here are correct. TS complains about method
+    // potentially being undefined, but in reality, this is fine (see docs).
+    const client = new StatsD({
+      host: statsdHost,
+      port: statsdPort,
+      prefix,
+      globalTags: {
+        path: ctx.path,
+        request_method: ctx.request.method,
+      },
+    });
+
+    // allow later middleware to add metrics
+    ctx.state.statsd = client;
+
+    const requestQueuingTime = ctx.request.get('X-Request-Start');
+    if (requestQueuingTime) {
+      try {
+        const timeAsInt = parseInt(requestQueuingTime, 10);
+        // @ts-ignore see description of tags in the hot-shots documentation
+        client.timing('request_queuing_time', timeAsInt, {});
+      } catch (err) {
+        // ignore errors here
```

Don't describe what ("ignoring errors"), describe why